src/IncludedCommands/UsefulFun/fly.lua

```diff
@@ -247,42 +267,101 @@ return {
                         for _,Event in pairs(self.Events) do
                             Event:Disconnect()
                         end
+
+                        --Unbind the ContextActionService bind
+                        ContextActionService:UnbindAction("toggleFlight")
     
                         --Unstore the flight.
                         if Api.CommandData.CurrentFlight == self then
                             Api.CommandData.CurrentFlight = nil
                         end
                     end
+
+                    ContextActionService:BindAction("toggleFlight",function(actionName, UserInputState, InputObject)
+                        -- Prevent multiple fires on 1 press
+                        if UserInputState ~= Enum.UserInputState.End then return end
+                        --Toggle the flight.
+                        if Flight.Active then
+                            (Flight :: any):Stop()
+                        else
+                            (Flight :: any):Start()
+                        end
+                    end,true,Enum.KeyCode.E,Enum.KeyCode.ButtonX)
+                    ContextActionService:SetTitle("toggleFlight","Toggle Flight")
+                    ContextActionService:SetPosition("toggleFlight",UDim2.new(0,50,0,100))
     
                     --Connect the events.
                     table.insert(Flight.Events, Humanoid.Died:Connect(function()
                         (Flight :: any):Destroy()
                     end))
                     table.insert(Flight.Events, UserInputService.InputBegan:Connect(function(Key, Processeed)
-                        if Processeed and Key.KeyCode ~= Enum.KeyCode.LeftShift and Key.KeyCode ~= Enum.KeyCode.RightShift then return end
+                        if Processeed and Flight.MultiPlatform[Key.KeyCode] ~= Enum.KeyCode.LeftShift and Flight.MultiPlatform[Key.KeyCode] ~= Enum.KeyCode.RightShift then return end
     
                         --Handle the key.
-                        if Key.KeyCode == Enum.KeyCode.E then
-                            --Toggle the flight.
-                            if Flight.Active then
-                                (Flight :: any):Stop()
-                            else
-                                (Flight :: any):Start()
-                            end
-                        elseif Flight.KeysDown[Key.KeyCode] ~= nil then
+                        if Flight.MultiPlatform[Key.KeyCode] ~= nil then
                             --Update the key press.
-                            Flight.KeysDown[Key.KeyCode] = true
+                            Flight.KeysDown[Flight.MultiPlatform[Key.KeyCode]] = true
                             Flight:UpdateMultipliers()
                         end
                     end))
+                    table.insert(Flight.Events, UserInputService.InputChanged:Connect(function(Key)
+                        local X,Y = nil,nil
```

Comment:
Again, spaces after commas.
```suggestion
                        local X, Y = nil, nil
```